deebotozmo/cli.py

```diff
@@ -0,0 +1,591 @@
+    await run_with_login(
+        cmd=CleanSpotArea,
+        cmd_dict={"area": rooms, "cleanings": cleanings},
+        device=ctx.obj["DEVICE"],
+    )
+
+
+@cli.command(name="setfanspeed", help="Set Clean Speed")
+@click.argument("speed", type=click.STRING, required=True)
+@click.pass_context
+@coro
+async def set_fan_speed(ctx: click.Context, speed: str) -> None:
+    """Click subcommand that sets the fan speed."""
+    await run_with_login(cmd=SetFanSpeed, cmd_list=[speed], device=ctx.obj["DEVICE"])
+
+
+@cli.command(name="setwaterlevel", help="Set Water Level")
+@click.argument("level", type=click.STRING, required=True)
+@click.pass_context
+@coro
+async def set_water_level(ctx: click.Context, level: str) -> None:
+    """Click subcommmand that sets the water level."""
+    await run_with_login(cmd=SetWaterLevel, cmd_list=[level], device=ctx.obj["DEVICE"])
+
+
+@cli.command(help="Returns to charger")
+@click.pass_context
+@coro
+async def charge(ctx: click.Context) -> None:
+    """Click subcommand that returns to charger."""
+    await run_with_login(cmd=Charge, device=ctx.obj["DEVICE"])
+
+
+@cli.command(help="Pause the robot")
+@click.pass_context
+@coro
+async def pause(ctx: click.Context) -> None:
+    """Click subcommand that pauses the clean."""
+    await run_with_login(cmd=CleanPause, device=ctx.obj["DEVICE"])
+
+
+@cli.command(help="Resume the robot")
+@click.pass_context
+@coro
+async def resume(ctx: click.Context) -> None:
+    """Click subcommand that resumes the clean."""
+    await run_with_login(cmd=CleanResume, device=ctx.obj["DEVICE"])
+
+
+@cli.command(name="getcleanlogs", help="Get Clean Logs")
+@click.pass_context
+@coro
+async def get_clean_logs(ctx: click.Context) -> None:
+    """Click subcommand that returns clean logs."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before(ctx.obj["DEVICE"])
+
+        event = asyncio.Event()
+
+        async def on_clean_event(clean_log_event: CleanLogEvent) -> None:
+            print(json.dumps(asdict(clean_log_event)))
+            event.set()
+
+        event.clear()
+        listener = vacbot.bot.events.clean_logs.subscribe(on_clean_event)
+        await event.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+@cli.command(help="Get robot statuses [Status,Battery,FanSpeed,WaterLevel]")
+@click.pass_context
+@coro
+async def statuses(ctx: click.Context) -> None:
+    """Click subcommand that returns the robot status."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before(ctx.obj["DEVICE"])
+        event = asyncio.Event()
+
+        async def on_status(status_event: StatusEvent) -> None:
+            print(f"Vacuum Available: {status_event.available}")
+            if isinstance(status_event.state, VacuumState):
+                print(f"Vacuum State: {status_event.state.name}")
+            event.set()
+
+        event.clear()
+        status_listener = vacbot.bot.events.status.subscribe(on_status)
```

Comment:
Question: Do you need that the different statues are printed in a fix order?
Otherwise it would improve the performance, when you subscribe (request all statues) in parallel